src/Http/Http.Results/src/Results.cs

```diff
@@ -742,15 +763,54 @@ public static IResult Problem(ProblemDetails problemDetails)
     /// <returns>The created <see cref="IResult"/> for the response.</returns>
     public static IResult ValidationProblem(
         IDictionary<string, string[]> errors,
+        string? detail,
+        string? instance,
+        int? statusCode,
+        string? title,
+        string? type,
+        IDictionary<string, object?>? extensions)
+    {
+        // TypedResults.ValidationProblem() does not allow setting the statusCode so we do this manually here
+        var problemDetails = new HttpValidationProblemDetails(errors)
+        {
+            Detail = detail,
+            Instance = instance,
+            Type = type,
+            Status = statusCode,
+        };
+
+        problemDetails.Title = title ?? problemDetails.Title;
+
+        CopyExtensions(extensions, problemDetails);
+
+        return TypedResults.Problem(problemDetails);
+    }
+
+    /// <summary>
+    /// Produces a <see cref="StatusCodes.Status400BadRequest"/> response
+    /// with a <see cref="HttpValidationProblemDetails"/> value.
+    /// </summary>
+    /// <param name="errors">One or more validation errors.</param>
+    /// <param name="detail">The value for <see cref="ProblemDetails.Detail" />.</param>
+    /// <param name="instance">The value for <see cref="ProblemDetails.Instance" />.</param>
+    /// <param name="statusCode">The status code.</param>
+    /// <param name="title">The value for <see cref="ProblemDetails.Title" />. Defaults to "One or more validation errors occurred."</param>
+    /// <param name="type">The value for <see cref="ProblemDetails.Type" />.</param>
+    /// <param name="extensions">The value for <see cref="ProblemDetails.Extensions" />.</param>
+    /// <returns>The created <see cref="IResult"/> for the response.</returns>
+#pragma warning disable RS0027 // Public API with optional parameter(s) should have the most parameters amongst its public overloads
+    public static IResult ValidationProblem(
+#pragma warning restore RS0027 // Public API with optional parameter(s) should have the most parameters amongst its public overloads
+        IEnumerable<KeyValuePair<string, string[]>> errors,
         string? detail = null,
         string? instance = null,
         int? statusCode = null,
         string? title = null,
         string? type = null,
-        IDictionary<string, object?>? extensions = null)
+        IEnumerable<KeyValuePair<string, object?>>? extensions = null)
     {
         // TypedResults.ValidationProblem() does not allow setting the statusCode so we do this manually here
-        var problemDetails = new HttpValidationProblemDetails(errors)
+        var problemDetails = new HttpValidationProblemDetails(new Dictionary<string, string[]>(errors ?? throw new ArgumentNullException(nameof(errors)), StringComparer.Ordinal))
```

Comment:
Why not `ArgumentNullException.ThrowIfNull(errors);` as above?

Reply:
Done